backend/src/api/petition-signature/controllers/petition-signature.js

```diff
@@ -0,0 +1,104 @@
+"use strict";
+
+/**
+ *  petition-signature controller
+ */
+
+const { createCoreController } = require("@strapi/strapi").factories;
+
+const path = require("path");
+
+function joinPaths(...args) {
+    return path.posix.join(...args).replace(':/', '://');
+}
+
+module.exports = createCoreController(
+  "api::petition-signature.petition-signature",
+  ({ strapi }) => ({
+    async add(ctx) {
+      const { petitionId, name, email, questionAnswers } = ctx.request.body;
+      if (!petitionId || !name || !email) {
+        ctx.throw(400, "Missing parameters");
+        return;
+      }
+      const petition = await strapi
+        .query("api::petition.petition")
+        .findOne({ where: { id: { $eq: petitionId } } });
+      if (!petition) {
+        ctx.throw(404, "petition not found");
+        return;
+      }
+
+      const petitionSignature = await strapi
+        .service("api::petition-signature.petition-signature")
+        .generateSignature(petition, name, email, questionAnswers);
+      const result = await strapi
+        .service("api::petition-signature.petition-signature")
+        .create({ data: petitionSignature });
+      if (!result) {
+        ctx.throw(500, "error generating signature");
+        return;
+      }
+
+      const exists = await strapi.entityService.count(
+        "api::petition-signature.petition-signature",
+        {
+          filters: {
+            petition: petitionId,
+            email: { $eq: email },
+          },
+        }
+      );
+
+      if (exists > 1) {
+        await strapi.entityService.delete(
+          "api::petition-signature.petition-signature",
+          result.id
+        );
+        ctx.send({ existed: true });
+        return;
+      }
+
+      const serverUrl = strapi.config.get('server.url', 'https://roodjongeren.nl/backend');
+      const confirmUrl = joinPaths(serverUrl, `/api/petition-signatures/confirm/${petitionSignature.confirmationCode}`);
+
+      await strapi.plugins["email"].services.email.send({
+        to: email,
+        subject: "Verifieer je email om deze petitie te ondertekenen",
+        text: `Ga naar deze link om je mail te verifieren: ${confirmUrl}`,
+        html: `Klik hier om je mail te verifieren: <a href="${confirmUrl}">Verifieer email</a>`,
+      });
+
+      ctx.send({ existed: false });
+    },
+
+    async confirm(ctx) {
+      const { confirmationCode } = ctx.params;
+
+      if (!confirmationCode) {
+        ctx.throw(403, "Onjuiste parameters");
```

Comment:
Kan deze in het Engels, of iig het consistent maken wanneer je NL of EN gebruikt voor error messages?